processor/spanmetricsprocessor/config_test.go

```diff
@@ -22,96 +22,79 @@ import (
+			},
 		},
 		{
-			configFile:          "config-full.yaml",
-			wantMetricsExporter: "otlp/spanmetrics",
-			wantLatencyHistogramBuckets: []time.Duration{
-				100 * time.Microsecond,
-				1 * time.Millisecond,
-				2 * time.Millisecond,
-				6 * time.Millisecond,
-				10 * time.Millisecond,
-				100 * time.Millisecond,
-				250 * time.Millisecond,
-			},
-			wantDimensions: []Dimension{
-				{"http.method", &defaultMethod},
-				{"http.status_code", nil},
+			id: component.NewIDWithName(typeStr, "full"),
+			expected: &Config{
+				MetricsExporter:        "otlp/spanmetrics",
+				AggregationTemporality: delta,
+				DimensionsCacheSize:    1500,
+				MetricsFlushInterval:   30 * time.Second,
+				LatencyHistogramBuckets: []time.Duration{
+					100 * time.Microsecond,
+					1 * time.Millisecond,
+					2 * time.Millisecond,
+					6 * time.Millisecond,
+					10 * time.Millisecond,
+					100 * time.Millisecond,
+					250 * time.Millisecond,
+				},
+				Dimensions: []Dimension{
+					{"http.method", &defaultMethod},
+					{"http.status_code", nil},
+				},
 			},
-			wantDimensionsCacheSize:    1500,
-			wantAggregationTemporality: delta,
-			wantMetricsFlushInterval:   30 * time.Second,
 		},
 	}
-	for _, tc := range testcases {
-		t.Run(tc.configFile, func(t *testing.T) {
-			// Prepare
-			factories, err := otelcoltest.NopFactories()
-			require.NoError(t, err)
-
-			factories.Receivers["otlp"] = otlpreceiver.NewFactory()
-			factories.Receivers["jaeger"] = jaegerreceiver.NewFactory()
-
-			factories.Processors[typeStr] = NewFactory()
-			factories.Processors["batch"] = batchprocessor.NewFactory()
 
-			factories.Exporters["otlp"] = otlpexporter.NewFactory()
-			factories.Exporters["prometheus"] = prometheusexporter.NewFactory()
-			factories.Exporters["jaeger"] = jaegerexporter.NewFactory()
+	for _, tt := range tests {
+		t.Run(tt.id.String(), func(t *testing.T) {
+			cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
+			require.NoError(t, err)
 
-			// Test
-			cfg, err := otelcoltest.LoadConfigAndValidate(filepath.Join("testdata", tc.configFile), factories)
+			factory := NewFactory()
+			cfg := factory.CreateDefaultConfig()
 
-			// Verify
+			sub, err := cm.Sub(tt.id.String())
 			require.NoError(t, err)
-			require.NotNil(t, cfg)
-			assert.Equal(t,
-				&Config{
-					MetricsExporter:         tc.wantMetricsExporter,
-					LatencyHistogramBuckets: tc.wantLatencyHistogramBuckets,
-					Dimensions:              tc.wantDimensions,
-					DimensionsCacheSize:     tc.wantDimensionsCacheSize,
-					AggregationTemporality:  tc.wantAggregationTemporality,
-					MetricsFlushInterval:    tc.wantMetricsFlushInterval,
-				},
-				cfg.Processors[component.NewID(typeStr)],
-			)
+			require.NoError(t, component.UnmarshalConfig(sub, cfg))
+
+			if tt.expected == nil {
+				assert.EqualError(t, component.ValidateConfig(cfg), tt.errorMessage)
+				return
+			}
+			assert.NoError(t, component.ValidateConfig(cfg))
```

Comment:
`tt.expected` is never `nil`, so we can remove this block and `tt.errorMessage`. Moreover, `component.ValidateConfig` already has its own set of test cases, so I don't think we need to worry about testing error cases there, which will simplify things for us.

I also suggest `require.NoError(t, component.ValidateConfig(cfg))`, since the below assertion would be invalid if `component.ValidateConfig(cfg)` returns an error.